custom_components/aseko_local/aseko_server.py

```diff
@@ -13,104 +13,205 @@
 
 
 class AsekoDeviceServer:
-    """Async TCP server for receiving and parsing Aseko unit data."""
+    """Async TCP server für Aseko-Gerätedaten."""
 
     _instances: ClassVar[dict[str, "AsekoDeviceServer"]] = {}
 
     def __init__(
         self,
         host: str = DEFAULT_BINDING_ADDRESS,
         port: int = DEFAULT_BINDING_PORT,
-        on_data: Callable[[AsekoDevice], None] | None = None,
+        on_data: Optional[Callable[[AsekoDevice], Any]] = None,
+        raw_sink: Optional[Callable[[bytes], Any]] = None,
     ) -> None:
-        """Initialize the server."""
         self.host = host
         self.port = port
         self.on_data = on_data
-        self._server: asyncio.AbstractServer | None = None
+        self._raw_sink = raw_sink
+        self._forward_cb: Optional[Callable[[bytes], Any]] = None
+        self._server: Optional[asyncio.AbstractServer] = None
+        self._clients: set[asyncio.StreamWriter] = set()
 
     async def start(self) -> None:
-        """Start the TCP server."""
+        """Start of the TCP server."""
 
         try:
             self._server = await asyncio.start_server(
                 self._handle_client, self.host, self.port
             )
-
-            _LOGGER.info("AsekoUnitServer started on %s:%d", self.host, self.port)
+            _LOGGER.info("AsekoDeviceServer gestartet auf %s:%d", self.host, self.port)
         except OSError as err:
-            _LOGGER.error("Failed to start AsekoUnitServer: %s", err)
-
-            message = f"Failed to start server: {err}"
-            raise ServerConnectionError(message) from err
+            _LOGGER.error("Serverstart fehlgeschlagen: %s", err)
+            raise ServerConnectionError(f"Failed to start server: {err}") from err
 
     async def stop(self) -> None:
-        """Stop the TCP server."""
-
-        if self._server is not None:
+        """Stop the TCP server and disconnect all clients."""
+
+        if self._server:
+            for w in list(self._clients):
+                try:
+                    w.close()
+                    await w.wait_closed()
+                except Exception:
+                    pass
+            self._clients.clear()
             self._server.close()
             await self._server.wait_closed()
-
-            _LOGGER.info("AsekoUnitServer stopped")
+            self._server = None
+            _LOGGER.info("AsekoDeviceServer gestoppt")
 
     @property
     def running(self) -> bool:
         """Check if the server is running."""
-
         return self._server is not None and self._server.is_serving()
 
+    async def _maybe_await(self, result: Any) -> None:
+        if asyncio.iscoroutine(result):
+            await result
+
+    async def _call_raw_sink(self, data: bytes) -> None:
+        if self._raw_sink:
+            try:
+                await self._maybe_await(self._raw_sink(data))
+            except Exception:
+                _LOGGER.debug("Raw sink raised an exception", exc_info=True)
```

Comment:
In case it is an error/exception, I would log it as it.
```suggestion
                _LOGGER.error("Raw sink raised an exception", exc_info=True)
```